guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,927 @@
+   |
+   = note: required because of the requirements on the impl of `diesel::query_builder::IntoUpdateTarget` for `&diesel_demo::models::User`
+   = note: required by `diesel::update`
+
+error[E0277]: the trait bound `diesel_demo::models::User: diesel::associations::HasTable` is not satisfied
+  --> src/bin/main.rs:37:5
+   |
+37 |     diesel::update(&hero).set(
+   |     ^^^^^^^^^^^^^^^^^^^^^ the trait `diesel::associations::HasTable` is not implemented for `diesel_demo::models::User`
+   |
+   = note: required because of the requirements on the impl of `diesel::associations::HasTable` for `&diesel_demo::models::User`
+
+continued stack trace...
+```
+The first error shows that the compiler is expecting to find a field `id` on the struct
+even though we wanted the method.
+The note associated with this error is very important,
+which tells us the `Identifiable` trait is not implemented for our `User` struct.
+
+The compiler is also giving us some more hints about trait bounds in each one of those *note* sections.
+The error messages regarding `IntoUpdateTarget` and `HasTable` are due to the trait bounds
+on the `update()` method that we called after `id()`.
+
+[Diesel API docs]: https://docs.diesel.rs/diesel/associations/trait.Identifiable.html
+
+## AsChangeset
+As we've seen with inserting data, updating data with tuples can also become tedious.
+`AsChangeset` serves a similar purpose to `Insertable` in that
+it provides us a way to easily use deserialized data from web forms for updates.
+To derive the `AsChangeset, add
+[`#[derive(AsChangeset)]`][aschangeset_doc] to your struct.
+
+In this section we will change our schema to make our `email` field nullable.
+`AsChangeset` allows us to deal with updating nullable fields a few different ways,
+which are changing the value, nulling he value, or ignoring the field.
+
+The nullable fields on our structs will now be of type `Option<T>`.
+Usually you do not want to change the primary key of the row or rows that you're updating.
+For this reason, we don't want to have `AsChangeset` and `Queryable` annotated on the same struct,
+which means we'll need another struct for updating our records.
+
+[aschangeset_doc]: https://docs.diesel.rs/diesel/query_builder/trait.AsChangeset.html
+
+Before we dive into some examples,
+let us take a look at our new schema.
+
+```rust
+// Output of "diesel print-schema"
+
+table! {
+    users (id) {
+        id -> Int4,
+        first_name -> Varchar,
+        last_name -> Varchar,
+        email -> Nullable<Varchar>,
+    }
+}
+```
+
+`AsChangeset` will automatically assume you don't want to change your record's primary key,
+so it will ignore that column.
+If your primary key field name is different than `id`,
+you must annotate your struct with the `#[primary_key(your_key)]` attribute.
+Our User struct now has an optional `email` field 
+(remember, nullable fields are Option<T>).
```

Comment:
I remember, you said so only a few sentences ago. But you still haven't shown me any `Option<_>`! So: I'd skip that remark here.